### PR TITLE
Adding the basic function of Keysight N5767A power supply

### PR DESCRIPTION
**Pull request by sumatrae**
Add keysight instruments N57xxA series power supply, including basic function such as  current/voltage and range, switch state

Signed-off-by: sumatrae <676119192@qq.com>

### CODE REVIEW

**Comment by cjermain** [comment]
@sumatrae, thanks for the PR!

In general, I would prefer to keep instruments scoped to specific models, and use class inheritance when necessary for sharing of properties. Can you target the N57xxA class to be the specific model you are testing it with?

**Comment by bilderbuchi** [comment]
@sumatrae there is no need to close this PR, you can just push updates/more commits to your originating branch (called sumatrae) on github, and this PR will be automatically updated!

**Comment by sumatrae** [comment]
hi @bilderbuchi, sorry for I am a new github guy.  Do you means PR will update all modifications from my master branch(sumatrae/pymeasure) automatically? If there are another modification, will it be updated in the same PR？

**Comment by sumatrae** [comment]
@cjermain @bilderbuchi and i have one more question that when I have added the source of an new instruments, should I set PR to sumatrae/pymeasure master at first?  Then set PR to ralph-group/pymeasure master ? Is it right?

**Comment by bilderbuchi** [comment]
You can see on the top ( sumatrae wants to merge 1 commit into ralph-group:master from sumatrae:sumatrae ) that your branch that this PR is created from is "sumatrae" in _your_ fork of pymeasure. This is targeted at the "master" branch from the ralph-group repo. This is both OK. 

If you now add more commits to your "sumatrae" branch, and push this to Github, this PR will automatically be updated with the new commits, they will appear in the conversation view and commits list etc, and the check will run again. This is the intended way to update/change/modify a PR after getting feedback or requests for modifications. 

The way to propose a PR is
* Choose a branch in the target repo that you want to target. Often this is `master`.
* Make sure you have the current version of that branch in your local repo.
* Make a new branch there to start work, give it an expressive name (e.g. instrument-<somename> or fix-<issuenr>) for your own sake
* Write the code, test, docs, etc. Follow https://pymeasure.readthedocs.io/en/latest/dev/adding_instruments.html#
* Push your new branch (which now has a couple of commits in it, on top of what was already in ralph-group/pymeasure:master) into your repo on github.
* Go to Github, make a PR (there will be already a prominent green button prompting you for this).
* Upon change requests, fix stuff locally, push again to your fork, the PR gets automatically updated.

Hope this helps :-)

File: docs/api/instruments/keysight/keysightN5767A.rst [hunk 1]
```diff
@@ -0,0 +1,9 @@
+########################
+keysight N5767A power supply
```
Comment (cjermain) [inline docs/api/instruments/keysight/keysightN5767A.rst]: Keysight should be capitalized here.

File: AUTHORS.txt [hunk 2]
```diff
@@ -15,3 +15,4 @@ Joseph Mittelstaedt
 Troy Fox
 Vikram Sekar
 Casper Schippers
+Sumatran Tiger
```
Comment (cjermain) [inline AUTHORS.txt]: Please also add yourself to `docs/about/authors.rst`.

File: pymeasure/instruments/keysight/keysightN5767A.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    )
+
+    voltage = Instrument.measurement("MEAS:VOLT?",
+        """ Reads a DC voltage measurement in Volts. """
+     )
+
+    ###############
+    # State (OFF/ON) #
+    ###############
+    state = Instrument.control(
```
Comment (cjermain) [inline pymeasure/instruments/keysight/keysightN5767A.py]: I would suggest providing explicit methods for `enable`, `disable`, and `is_enabled` in the same way that it is done in the [Danfysik8500 power supply](https://github.com/ralph-group/pymeasure/blob/master/pymeasure/instruments/danfysik/danfysik8500.py#L99-L112). The strings `ON` and `OFF` are understandable (especially given their direct use in the command), but they seem arbitrary relative to other instrument classes in PyMeasure.

File: pymeasure/instruments/keysight/keysightN5767A.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        # Set up data transfer format
+        if isinstance(self.adapter, VISAAdapter):
+            self.adapter.config(
+                is_binary=False,
+                datatype='float32',
+                converter='f',
+                separator=','
+            )
+
+    # TODO: Clean up error checking
```
Comment (cjermain) [inline pymeasure/instruments/keysight/keysightN5767A.py]: Is this TODO outstanding? If so, what else needs to be addressed?

File: AUTHORS.txt (hunk 2 above)
Comment (sumatrae) [reply AUTHORS.txt]: Do I need add myself to AUTHORS.txt ? Will contributor be added to AUTHORS.txt  by owner when related version release?

File: pymeasure/instruments/keysight/keysightN5767A.py (hunk 3 above)
Comment (sumatrae) [reply pymeasure/instruments/keysight/keysightN5767A.py]: I have added enable,disable,is_enabled, and add _status measurement to read power supply status.
As you said ,so users may not care about string "ON"/"OFF“ or 0,1, returns type are boolean.

**Comment by cjermain** [comment]
Looks good! Thanks @sumatrae.

**Pull request merged**